target/linux/x86/image/64.mk

```diff
@@ -4,11 +4,21 @@ define Device/generic
   DEVICE_PACKAGES += \
 	kmod-amazon-ena kmod-amd-xgbe kmod-bnx2 kmod-e1000e kmod-e1000 \
 	kmod-forcedeth kmod-fs-vfat kmod-igb kmod-igc kmod-ixgbe kmod-r8169 \
-	kmod-tg3 kmod-mlxsw-core kmod-mlxsw-pci kmod-mlxsw-i2c \
+	kmod-tg3
+  GRUB2_VARIANT := generic
+endef
+TARGET_DEVICES += generic
+
+define Device/nvidia_spectrum
```

Comment:
I don't think, that we want to start with per device rootfs. I would probably go with something more generic, like `Device/large` so we could add more stuff into that image, like `usbutils` etc. see https://lists.openwrt.org/pipermail/openwrt-devel/2023-November/041767.html for more details.

Cc: @aparcar @blocktrron

Reply:
Scratch the above.

Looking into this further https://elixir.bootlin.com/linux/v6.1.62/source/drivers/net/ethernet/mellanox/mlxsw/spectrum.c#L62 we can see following:

```
#define MLXSW_SP_FWREV_MINOR 2010
#define MLXSW_SP_FWREV_SUBMINOR 1006
#define MLXSW_SP1_FWREV_MAJOR 13

#define MLXSW_SP1_FW_FILENAME \
        "mellanox/mlxsw_spectrum-" __stringify(MLXSW_SP1_FWREV_MAJOR) \
        "." __stringify(MLXSW_SP_FWREV_MINOR) \
        "." __stringify(MLXSW_SP_FWREV_SUBMINOR) ".mfa2"
```

so the kernel is going to load single firmware file `mlxsw_spectrum-13.2010.1006.mfa2`? And we could simply just ship that single file and save 29MiB?

Reply:
> I don't think, that we want to start with per device rootfs. I would probably go with something more generic, like `Device/large` so we could add more stuff into that image, like `usbutils` etc. see https://lists.openwrt.org/pipermail/openwrt-devel/2023-November/041767.html for more details.
> 
> Cc: @aparcar @blocktrron

agree with this, but my idea is more radical. It is better to only retain support for x86_64.
I don't know how many people are still using 32-bit x86, this needs to be discussed.

Reply:
> Scratch the above.
> 
> Looking into this further https://elixir.bootlin.com/linux/v6.1.62/source/drivers/net/ethernet/mellanox/mlxsw/spectrum.c#L62 we can see following:
> 
> ```
> #define MLXSW_SP_FWREV_MINOR 2010
> #define MLXSW_SP_FWREV_SUBMINOR 1006
> #define MLXSW_SP1_FWREV_MAJOR 13
> 
> #define MLXSW_SP1_FW_FILENAME \
>         "mellanox/mlxsw_spectrum-" __stringify(MLXSW_SP1_FWREV_MAJOR) \
>         "." __stringify(MLXSW_SP_FWREV_MINOR) \
>         "." __stringify(MLXSW_SP_FWREV_SUBMINOR) ".mfa2"
> ```
> 
> so the kernel is going to load single firmware file `mlxsw_spectrum-13.2010.1006.mfa2`? And we could simply just ship that single file and save 29MiB?

It is not difficult to package the specified version. Do we still need auto update the firmware to asic when the system starts? Other systems with these devices have done this.
But these systems have full chassis management function, and we clearly have no incentive to implement these? (The simplest fan temperature control can be solved using sensors and pwmconfig)

Reply:
> > I don't think, that we want to start with per device rootfs. I would probably go with something more generic, like `Device/large` so we could add more stuff into that image, like `usbutils` etc. see https://lists.openwrt.org/pipermail/openwrt-devel/2023-November/041767.html for more details.
> > Cc: @aparcar @blocktrron
> 
> agree with this, but my idea is more radical. It is better to only retain support for x86_64. I don't know how many people are still using 32-bit x86, this needs to be discussed.

If you want to reply to messages on the mailing list, perhaps you should reply to the mailing list?  Otherwise people who were interested in the original message won't ever see the reply.

No one advocated against reducing the number of x86 builds by one.  Removing all x86 seems a harder sell.  Systems with Geode processor were available for purchase [into 2021](https://www.pressebox.com/pressrelease/congatec-ag/congatec-partners-with-AMD-for-long-time-support-of-AMD-Geode-processors/boxid/888930), so those seem worthy of some continued support.  Most desktop Linux distributions continue to support i686, yet I'm unsure these are a good fit for OpenWRT.

Perhaps, but please reply to the mailing list so everyone who read the original message can see the follow-on discussion.